watershed_workflow/land_cover_properties_Soumendra.py

```diff
@@ -0,0 +1,328 @@
+        plt.show()
+
+    crosswalk = dict()
+    for i, nlcd in enumerate(unique_nlcd):
+        crosswalk[nlcd] = unique_modis[np.argmax(correlation_matrix[i])]
+    return crosswalk
+                                  
+def write_to_ats(lai_raster, crosswalk, smooth=False, smooth_filter=False, nyears=None):
+    """Accepts a numpy array of LAI data and returns a dictionary ATS data.
+    
+    Parameters:
+        lai_raster : dict
+            MODIS LAI dictinary
+        smooth : bool
+            whether to smooth the data or not
+        smooth_filter : bool
+            whether to apply a savgol filter for smoothing. Default is False.
+        nyears : int
+            how many years to repeat the typical year. Default is to get the same length as the raw data.
+        crosswalk : Maps keys of MODIS indices into NLCD indices.
+    """
+
+    outputs['modis_filename'] = f'../data-processed/{name}_MODIS_LAI_smoothed_2002_2020.h5'
+
+    df_smooth['time [s]'] = df_smooth['time [d]']*86400
+    with h5py.File(outputs['modis_filename'],'w') as fid:
+        for k in df_smooth:
+            fid.create_dataset(k, data=df_smooth[k][:])
+    
+    # for the entire 40 years data
+    outputs['modis_typical_filename'] = f'../data-processed/{name}_MODIS_LAI_typical_1980_2020.h5'
+
+    with h5py.File(outputs['modis_typical_filename'],'w') as fid:
+        for k in df_rep:
+            fid.create_dataset(k, data=df_rep[k][:])
+
+#
+#    nlcd_LAI['time [s]'] = (nlcd_LAI.index - nlcd_LAI.index[0]).total_seconds()
+#
+#    with h5.File(f'../data-raw/{name}_MODIS_LAI_072002_122020.h5', 'w') as fout:
+#
+#        for i in ['time [s]', f'NLCD {NLCDLULC1label1} LAI [-]']:
+#            fout.create_dataset(i, data= nlcd_LAI[i].values)
+            
+
+def plot_histogram(modis_lc):
+    """Plot histogram based on the MODIS LC pixels.
+
+    Parameters
+    ----------
+#    nlcd_lc : np.ndarray[NX,NY]
+#      Raster of a single year of NLCD land cover type.
+    modis_lc : np.ndarray[NX,NY]
+        Raster of a single year of MODIS land cover type.
+    """
+
+#    unique = ...
+#    plt.figure()
+#    plt.histogram()
+#    plt.show()
+    #colors = [lc_colors[i][1] for i in ids]
+    #lc_colors = [lc_type1_colors[i][1] for i in ids]
+
+    #labels = [lc_type1_colors[i][0] for i in ids]
+
+    #labelsp1 = [lc_type1_colors[i][0] for i in lc_type1_colors]
+
+    counts, bins = np.histogram(modis_lc, range=[-1,17], bins=18)
+    plt.hist(bins[:-1], bins=18, range=[-1,17], weights=counts)
+    plt.xlabel("MODIS LULC")
+    plt.ylabel("Counts")
+    plt.title("Histogram of number of pixels per LULC types")
+    plt.xticks(bins,labelsp1,rotation=90)
+    
+def area_cutoff(modis_lc):
+    
+    ids, counts = np.unique(modis_lc.values[~np.isnan(modis_lc.values)], return_counts=True)
+
+    sum1 = sum(counts)
+    cutoff = sum1*0.05 #Separate LULC types with 5% of the pixel coverage cutoff
+    
+
+
+################ Implementation in notebook ###############
```

Comment:
These sort of notes are really useful -- put them in the top level doc of the whole file.

So the first line of the file can be:

```
"""Computes land-cover and LAI properties.

## Usage in a notebook

1. modis and nlcd crosswalk comparison to enable users making their own judgement
2. logics related to LC type selection (1, 2, 3, 4 etc.)
3. plots of MODIS LC comparison with NLCD, MODIS LAI time-series
4. filter LAI data to remove the erroneously high and low values (-ve)
"""
```

This becomes file-level documentation, and gets printed when you do for instance, in the notebook:

```
import watershed_workflow.land_cover_properties
watershed_workflow.land_cover_properties?
```